File: 1-2.php
<?php

$filename = 'pandas/edu_bigdata_imp1.csv';

$csvData = array_map('str_getcsv', file($filename));
$header = array_shift($csvData);

$pseudoIDIndex = array_search('PseudoID', $header);
$questionSNIndex = array_search('dp001_question_sn', $header);

$uniqueValues = [];

foreach ($csvData as $row) {
    if ($row[$pseudoIDIndex] == 39 && isset($row[$questionSNIndex])) {
        $uniqueValues[] = $row[$questionSNIndex];
    }
}

$uniqueCount = count(array_unique($uniqueValues));

echo $uniqueCount;

?>

File: 1-2.py
import pandas as pd

df = pd.read_csv('pandas\edu_bigdata_imp1.csv', encoding = 'big5', low_memory = False)
df_filtered = df[df['PseudoID'] == 39]
df_filtered = df_filtered.dropna(subset = ['dp001_question_sn'])
unique_value = df_filtered['dp001_question_sn'].unique()
print(len(unique_value))

